examples/sign-leaf-with-ca.rs

```diff
@@ -0,0 +1,99 @@
+use rcgen::{
+	date_time_ymd, BasicConstraints, Certificate, CertificateParams, CertificateSigningRequest,
+	DnType, ExtendedKeyUsagePurpose, IsCa, KeyUsagePurpose,
+};
+use time::{Duration, OffsetDateTime};
+use x509_parser::certification_request::X509CertificationRequest;
+use x509_parser::prelude::FromDer;
+
+/// Example demonstrating signing end-endity certificate with ca
+fn main() {
+	let ca = Ca::new();
+	let entity = Entity::new();
+
+	println!("directly signed end-entity certificate:");
+	let direct = entity
+		.certificate
+		.serialize_pem_with_signer(&ca.certificate)
+		.unwrap();
+	println!("{direct}");
+
+	println!("ca certificate:");
+	let pem = ca.certificate.serialize_pem().unwrap();
+
+	println!("{}", pem);
+}
+
+struct Ca {
+	certificate: Certificate,
+}
+
+impl Ca {
+	fn new() -> Self {
+		let mut params = CertificateParams::new(vec!["ca.some.host".to_owned()]);
+		let (yesterday, tomorrow) = y_t();
+		params.is_ca = IsCa::Ca(BasicConstraints::Unconstrained);
+		params.distinguished_name.push(DnType::CountryName, "BR");
+		params
+			.distinguished_name
+			.push(DnType::OrganizationName, "Crab widgits SE");
```

Comment:
Both the country name and organization name should be using `DnValue::PrintableString` values based on the ASN.1 module in RFC 5280. I think using this form is resulting in `UTF8String` values.

Try:

```rust
params
			.distinguished_name
			.push(DnType::CountryName, DnValue::PrintableString("BR".into()));
```

Reply:
> Both the country name and organization name should be using DnValue::PrintableString

Ah, my mistake, just `CountryName` needs to be a `PrintableString`.

Org name is a `CHOICE` that allows `utf8String` so can be left as-is, but country name should be changed to a printable string [based on 5280 A.1](https://www.rfc-editor.org/rfc/rfc5280#appendix-A.1):

```
-- Naming attributes of type X520countryName (digraph from IS 3166)

id-at-countryName       AttributeType ::= { id-at 6 }

X520countryName ::=     PrintableString (SIZE (2))
```